fitbenchmarking/parsing/fitbenchmark_parser.py

```diff
@@ -309,6 +312,48 @@ def _get_starting_values(self):
 
         return starting_values
 
+    def _get_mantid_starting_values(self):
+        """
+        Get the starting values for the problem
+
+        :return: Starting values for the function
+        :rtype: list of OrderedDict
+        """
+        # Mantid functions can have reserved keywords so ignore these
+        ignore = ['name', 'ties', 'constraints']
+
+        name_template = '{1}' if len(self._parsed_func) == 1 else 'f{0}_{1}'
+        starting_values = []
+        for i, f in enumerate(self._parsed_func):
+
+            # use mantid function factory to seperate attributes and parameters
+            fcopy = f.copy()
+            if 'ties' in fcopy:
+                fcopy.pop('ties')
+            fstring = ", ".join(f"{key}={value}"
+                                for key, value in fcopy.items())
+            ffun = msapi.FunctionFactory.createInitialized(fstring)
+            attr_names = [s for s in ffun.attributeNames()]
+
+            # filter parameters
+            params = f.copy()
+            for key in ignore:  # filter reserved keywords from params
+                if key in params:
+                    params.pop(key)
+            for attr in attr_names:  # filter attributes from params
+                if attr in params:
+                    params.pop(attr)
+            if 'ties' in f:  # filter tied params from params
+                for tie in f['ties']:
+                    if tie in params:
+                        params.pop(tie)
+            starting_values += [(name_template.format(i, name), val)
+                                for name, val in params.items()]
```

Comment:
I'm a bit worried that this creates a badly ordered parameter list in the mantid -> non-mantid case.

i.e. `function=some_func,p0=1.0,p1=2.0,p2=4.0,ties=(p1=2.0)`
would lead to `starting_values == {'p0':1.0, 'p2':4.0}`
which then leads to `initial_params == [1.0, 4.0]` in the controllers...

I think this is covered in the [docs](https://fitbenchmarking.readthedocs.io/en/latest/users/problem_definition_files/native.html) under "Note: Any non-standard arguments (e.g. ties, constraints, fixes, …) will only work with Mantid fitting software. Using other minimizers to fit these problems will result in the non-standard arguments being ignored." which we should change if we're supporting it now.

Looking into this a bit more I'm concerned that if a user passes in:
`function=some_func,p0=1.0,p2=4.0,p1=2.0`
we'll get `initial_params == [1.0, 4.0, 2.0]` in the controller which could come out wrong... so maybe it's already broken 😅

Reply:
Yes looks like it's already broken, we should create a separate bug for this.